### PR TITLE
Issue #243 Show if Cluster Capacity is Reached

### PR DESCRIPTION
**Pull request by kishansagathiya**
Idler would be giving 503 if Cluster Tenant Resource Capacity is
Reached. Modified Unidle function to handle that

Now waiting template will show the message "Cluster Tenant resource
capacity for current cluster is full, please wait...", if
Cluster Capacity is reached

#243

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @kishansagathiya for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin - for example exclude certain file types so if PR contains only them it won't be checked.

File: internal/clients/idler.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ type clusterView struct {
 // IdlerService provides methods to talk to the idler client
 type IdlerService interface {
 	IsIdle(tenant string, openShiftAPIURL string) (bool, error)
-	UnIdle(tenant string, openShiftAPIURL string) error
+	UnIdle(tenant string, openShiftAPIURL string) (*int, error)
```
Comment (sthaha) [inline internal/clients/idler.go]: We should be using an `int` instead of `int * ` here. 
I am not sure how a pointer to a const is even going to work.

**Comment by kishansagathiya** [comment]
@sthaha @containscafeine 
What would be really nice for this is some UI magic such that the message that we see while jenkins is unidling changes dynamically i.e., If the ajax request that the page sent to proxy gives 503 change the message to `Cluster Tenant resource capacity for current cluster is full, please wait...`
and if it gives 202 it should change to `Jenkins has been idled. It is starting now, please wait...`

WDYT?

File: internal/proxy/proxy.go [hunk 2]
```diff
@@ -318,12 +318,12 @@ func (p *Proxy) handleJenkinsUIRequest(w http.ResponseWriter, r *http.Request, r
...
 				}
 				if isIdle { //If jenkins is idled, return loading page and status 202
-					err = p.idler.UnIdle(ns, clusterURL)
+					code, err := p.idler.UnIdle(ns, clusterURL)
 					if err != nil {
 						p.HandleError(w, err, requestLogEntry)
 						return
 					}
-					err = p.processTemplate(w, ns, requestLogEntry)
+					err = p.processTemplate(w, ns, requestLogEntry, code)
```
Comment (containscafeine) [inline internal/proxy/proxy.go]: we are not handling the error here

File: internal/proxy/proxy.go [hunk 3]
```diff
@@ -339,7 +339,7 @@ func (p *Proxy) handleJenkinsUIRequest(w http.ResponseWriter, r *http.Request, r
 						cookie.Expires = time.Unix(0, 0)
 						http.SetCookie(w, cookie)
 					} else {
-						err = p.processTemplate(w, ns, requestLogEntry)
+						err = p.processTemplate(w, ns, requestLogEntry, http.StatusAccepted)
```
Comment (containscafeine) [inline internal/proxy/proxy.go]: not handling here as well

Comment (kishansagathiya) [reply internal/proxy/proxy.go]: Addressed in latest commit

**Comment by containscafeine** [comment]
Nice work @kishansagathiya

**Pull request merged**